### ISSUE TITLE
LC0078 and TableType = Temporary

### ISSUE THREAD

**Issue by tscottjendev** [StefanMaron/BusinessCentral.LinterCop#1214]
It appears that [LC0078](https://github.com/StefanMaron/BusinessCentral.LinterCop/wiki/LC0078#temporary-records-should-not-trigger-table-triggers) recognizes the table property `TableType = Temporary;` but it seems inconsistent compared to a `Normal` table.

Normal TableType records declared as Temporary  -> warnings created, **correct!**
Temporary TableType records declared with Temporary  -> no warnings created, **Inconsistent!** should be warnings.
Temporary TableType records declared without Temporary -> no warnings created, but it would seem LinterCop knows the table type given the differences above, so shouldn't it give the same warnings and TableType Normal declared as Temporary

Given:
```
table 50100 MyTable
{
    TableType = Temporary;  //TEMPORARY 

    fields
    {
        field(1; "MyField"; Integer)
        {
            Caption = 'MyField';
        }
    }
}
```

```
table 50101 MyOtherTable
{
  //NORMAL table
    fields
    {
        field(1; "MyField"; Integer)
        {
            Caption = 'MyField';
        }
    }

```

In this procedure I get different warnings.
```
    procedure MyProcedure()
    var
        myRecord: Record MyTable;
        TempMyRecord: Record MyTable temporary;
        TempMyOtherRecord: Record MyOtherTable temporary;
    begin
        // Standard declaration 
        MyRecord.Validate(MyField, 2);  //No Warning
        MyRecord.Modify(true); //No Warning

        // Temporary TableType declared as temporary
        TempMyRecord.Validate(MyField, 2);  //No Warning
        TempMyRecord.Modify(true); //No Warning

        // Normal TableType declared as temporary
        TempMyOtherRecord.Validate(MyField, 2);  //Warning: Temporary records should not trigger the table triggers. (LC0078)
        TempMyOtherRecord.Modify(true); //Warning: Temporary records should not trigger the table triggers. (LC0078)
    end;
```

**Comment by rvanbekkum** [comment]
I think LC0078 should only be raised for record variables with TableType = Normal.

Otherwise there would be no point in even having triggers in a TableType=Temporary table.
The triggers in TableType =Temporary tables are/should have been implemented with 'temporary' use in mind.

TableType = Normal table record variables is where the risk is typically.
Often developers of the TableType = Normal (and developers that hook into the validation triggers, e.g., via subscribers) did not properly take into account 'temporary use' for the table (and for some tables it would be very difficult to).

Examples are the 'Sales Line' or 'Item' table, where validation triggers might trigger inserts/modifications/deletions of related tables even when used as temporary record variables.
Actually, it's risky to invoke TempSalesLine.Insert/Modify/Delete(false) alltogether, as 'RunTriggrr=false' event subscribers from a third party that did not properly take into account 'IsTemporary' might be changing actual data.
(Could therefore be even good to consider raising a diagnostic for TempNormalTableRec.Insert(false) to discourage that when it's used with a Record variable of a TableType = Normal table)

**Comment by Arthurvdv** [comment]
Hi @tscottjendev and/or @rvanbekkum 

Thanks for the detailed report and indeed the behavior is inconsistent. Not only on this rule, but I'v found several rules that would check if the property is explicitly added to the object, but didn't look at the value itself. So `TableType = Temporary;` and `TableType = Normal;` would both return `true` that the property `TableType` is set (or something equivalent).

I believe this is already resolved in the ALCops project, where the equivalent rule (PC0027) properly accounts for both scenarios, detecting temporary records regardless of whether the temporariness comes from the variable declaration or the table's TableType property.

I'm closing this issue here since this repository is deprecated and no longer receiving updates (as noted in the README). Development continues at ALCops.

If you haven't already, I'd invite you to look into migrating to ALCops. The documentation is at alcops.dev, and a migration guide at alcops.dev/docs/lintercop-migration (https://alcops.dev/docs/lintercop-migration/).

If you find that this still not working correctly, please feel free to open a new issue at ALCops/Analyzers (https://github.com/ALCops/Analyzers/issues) and I'll address it there.
